### PR TITLE
Use rpm 4.15 in RPM builders to get FIPS-installable RPMs

### PR DESCRIPTION
**Pull request by bkabrda**

### CODE REVIEW

File: patches/rpm-4.15.1-fix-rpmbuild-segfault.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff --git a/build/pack.c b/build/pack.c
```
Comment (mx-psi) [inline patches/rpm-4.15.1-fix-rpmbuild-segfault.patch:1]: Maybe add a comment about the need and purpose of the patch? Either in the diff file (I think the first lines get ignored for example) or in the PR description

**Review by mx-psi — APPROVED** [review]

**Review by albertvaka — APPROVED** [review]

File: rpm-x64/Dockerfile [hunk 2]
```diff
@@ -27,12 +27,56 @@ RUN sed -is 's/enabled=0/enabled=1/g' /etc/yum.repos.d/CentOS-Vault.repo && \
...
+# (which makes `rpmbuild --sign` succeed even without tty)
+RUN curl -sL -o /tmp/rpm-4.15.1.tar.bz2 http://ftp.rpm.org/releases/rpm-4.15.x/rpm-4.15.1.tar.bz2 \
+    && cd /tmp \
+    && tar -xjf /tmp/rpm-4.15.1.tar.bz2 \
+    && cd rpm-4.15.1 \
+    && cat /tmp/rpm-4.15.1-fix-rpmbuild-segfault.patch | patch -p1 \
+    && ./configure --without-lua --without-audit \
+    && make \
+    && make install \
+    && sed -i -e 's/ <.*/\x27 \\/' /usr/local/lib/rpm/rpmpopt-4.15.1 \
```
Comment (albertvaka) [inline rpm-x64/Dockerfile:72]: Is this fix something you came up with or is it used commonly? I would expect that running `rpmbuild --sign` in a CI is a common use case, so I'm surprised the change is even needed.

Comment (albertvaka) [reply rpm-x64/Dockerfile:72]: Ok, I see it's documented at least here: https://bugzilla.redhat.com/show_bug.cgi?id=1239039

File: rpm-arm64/Dockerfile [hunk 3]
```diff
@@ -22,9 +22,36 @@ ENV CONDA_PATH /root/miniforge3
...
+    bzip2-devel e2fsprogs-devel file-devel libacl-devel libarchive-devel libattr-devel \
+    libxml2-devel lzo-devel nss nss-devel popt-devel sharutils xz-devel
+
+# Build new rpm
+COPY patches/rpm-4.15.1-fix-rpmbuild-segfault.patch /tmp
+# The sed command basically replaces this:
+#   --pipe 'rpm --addsign `grep ".*: .*\.rpm$"|cut -d: -f2` < "/dev/"`ps -p $$ -o tty | tail -n 1`' \
+# (which makes `rpmbuild --sign` fail when there's no tty - e.g. in Gitlab) with this:
+#   --pipe 'rpm --addsign `grep ".*: .*\.rpm$"|cut -d: -f2`' \
+# (which makes `rpmbuild --sign` succeed even without tty)
```
Comment (albertvaka) [inline rpm-arm64/Dockerfile:38]:
```suggestion
# (which makes `rpmbuild --sign` succeed even without tty)
# See: https://bugzilla.redhat.com/show_bug.cgi?id=1239039
```

Comment (albertvaka) [inline rpm-arm64/Dockerfile:38]:
```suggestion
# (which makes `rpmbuild --sign` succeed even without tty)
# See: https://bugzilla.redhat.com/show_bug.cgi?id=1239039
```

File: rpm-x64/Dockerfile [hunk 4]
```diff
@@ -27,12 +27,56 @@ RUN sed -is 's/enabled=0/enabled=1/g' /etc/yum.repos.d/CentOS-Vault.repo && \
...
+    && rpm -Uvh --nodeps /root/rpmbuild/RPMS/x86_64/* \
+    && rm -rf /tmp/libarchive-3.1.2-12.el7.src.rpm /root/rpmbuild
+
+# Actually build new rpm
+COPY patches/rpm-4.15.1-fix-rpmbuild-segfault.patch /tmp
+# The sed command basically replaces this:
+#   --pipe 'rpm --addsign `grep ".*: .*\.rpm$"|cut -d: -f2` < "/dev/"`ps -p $$ -o tty | tail -n 1`' \
+# (which makes `rpmbuild --sign` fail when there's no tty - e.g. in Gitlab) with this:
+#   --pipe 'rpm --addsign `grep ".*: .*\.rpm$"|cut -d: -f2`' \
+# (which makes `rpmbuild --sign` succeed even without tty)
```
Comment (albertvaka) [inline rpm-x64/Dockerfile:63]:
```suggestion
# (which makes `rpmbuild --sign` succeed even without tty)
# See: https://bugzilla.redhat.com/show_bug.cgi?id=1239039
```

File: rpm-arm/Dockerfile [hunk 5]
```diff
@@ -21,15 +21,44 @@ ENV DD_TARGET_ARCH $DD_TARGET_ARCH
...
+    bzip2-devel e2fsprogs-devel file-devel libacl-devel libarchive-devel libattr-devel \
+    libxml2-devel lzo-devel nss nss-devel popt-devel sharutils xz-devel
+
+# Build new rpm
+COPY patches/rpm-4.15.1-fix-rpmbuild-segfault.patch /tmp
+# The sed command basically replaces this:
+#   --pipe 'rpm --addsign `grep ".*: .*\.rpm$"|cut -d: -f2` < "/dev/"`ps -p $$ -o tty | tail -n 1`' \
+# (which makes `rpmbuild --sign` fail when there's no tty - e.g. in Gitlab) with this:
+#   --pipe 'rpm --addsign `grep ".*: .*\.rpm$"|cut -d: -f2`' \
+# (which makes `rpmbuild --sign` succeed even without tty)
```
Comment (albertvaka) [inline rpm-arm/Dockerfile:45]:
```suggestion
# (which makes `rpmbuild --sign` succeed even without tty)
# See: https://bugzilla.redhat.com/show_bug.cgi?id=1239039
```

File: rpm-x64/Dockerfile (hunk 2 above)
Comment (bkabrda) [reply rpm-x64/Dockerfile:72]: Actually, in the latest version of my [omnibus PR](https://github.com/DataDog/omnibus-ruby/pull/115), I find out that `rpmbuild --sign` is still broken for packages with lots of files. Due to the fact that it's also deprecated, I replaced it with calling `rpm --addsign` after doing the build. This means that I can now remove this `sed` call.

**Review by albertvaka — APPROVED** [review]

**Review by albertvaka — APPROVED** [review]

**Pull request merged**